### DISCUSSION TITLE
Is it possible to generate substructures which are valid structures for a molecule?

### DISCUSSION THREAD

**Discussion by manajit-das** [rdkit/rdkit#5761] [Q&A]
Can I get all the possible substructure which are chemically valid molecule enumerating from all the atoms in a molecule . For instance the fingerprint generation process creates all sorts of fragments, but the fragments are not chemically valid molecule. 
Thanks for your kind help

**Comment by greglandrum** [comment]
@manajit-das, I don't understand the question. What do you mean by "chemically valid molecule"? Can you provide a few examples of what you mean by valid and invalid molecules from substructures?

**Reply by manajit-das** [reply]
Sorry for the poor construction of the question. 
Let me try with an example, For the molecule N-methyl-1-phenylmethanamine, a circular fragment of radius=1 around the N atom will give dimethyl amine which is a valid molecule, whereas that for radius=3 around the N atom will result a fragment which will break the aromatic ring and the fragment is not a valid molecule, So is it possible that whenever such scenario appears it will return the molecule by breaking bonds which doesn't disturb the aromaticity or the ring.  
![fragmentation](https://user-images.githubusercontent.com/59758360/201903007-c48ce564-a3a5-4c22-890b-04069dfa6a26.png)
I hope it is okay, please let me know. Thank you.
